docs/orchestrating-elastic-stack-applications/stack-config-policy.asciidoc

```diff
@@ -0,0 +1,236 @@
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html[Role mappings]
+
+A policy can be applied to one or more Elasticsearch clusters in any namespace managed by the ECK operator.
+
+Configuration policy settings applied by the ECK operator are immutable through the Elasticsearch REST API.
+
+It is currently not allowed to configure an Elasticsearch cluster with more than one policy.
+
+[float]
+[id="{p}-{page_id}-definition"]
+== Define Elastic Stack configuration policies
+
+Elastic Stack configuration policies can be defined in a `StackConfigPolicy` resource. Each `StackConfigPolicy` must have the following fields:
+
+* `name` is a unique name used to identify the policy.
+* `spec.elasticsearch` describes the settings to configure and at least one setting must be defined. Each of the following fields is a named entity map:
+  ** `clusterSettings` are the settings commonly found in the elasticsearch.yaml file.
+  ** `snapshotRepositories` are snapshot repositories for defining an off-cluster storage location for your snapshots.
+  ** `snapshotLifecyclePolicies` are snapshot lifecycle policies, to automatically take snapshots and control how long they are retained.
+  ** `securityRoleMappings` are role mappings, to define which roles are assigned to each user by identifying them through rules.
+  ** `ingestPipelines` are ingest pipelines, to perform common transformations on your data before indexing.
+  ** `indexLifecyclePolicies` are index lifecycle policies, to automatically manage the index lifecycle.
+  ** `indexTemplates.componentTemplates` are component templates that are building blocks for constructing index templates that specify index mappings, settings, and aliases.
+  ** `indexTemplates.composableIndexTemplates` are index templates to define settings, mappings, and aliases that can be applied automatically to new indices.
+
+The following fields are optional:
+
+* `namespace` is the namespace of the `StackConfigPolicy` resource and used to identify the Elasticsearch clusters to which this policy applies. If it equals to the operator namespace, all namespaces managed by the operator are used, otherwise only the policy namespace is used.
+* `resourceSelector` is a link:https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/[label selector] to identify the Elasticsearch clusters to which this policy applies in combination with the namespace(s). No `resourceSelector` means all Elasticsearch clusters in the namespace(s).
+* `secureSettings` is a list of Secrets containing Secure Settings to inject into the keystore(s) of the Elasticsearch cluster(s) to which this policy applies, similar to the<<{p}-custom-images,Elasticsearch Secure Settings>>.
+
+Secure settings are required to configure Cloud snapshot repositories (azure, gcs, s3) if you are not using Workload Identity (e.g.: https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity).
```

Comment:
AWS IAM roles for service accounts is a similar feature on AWS that allows users to avoid having to specify secure settings. https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-snapshots.html#k8s-iam-service-accounts

Reply:
I thought "Workload Identity" might be the generic term for this feature but not really and I forgot we already documented it so well. I will make the link between the two docs.

```suggestion
Secure settings are required to configure Cloud snapshot repositories (azure, gcs, s3) if you are not using Cloud-provider specific means to leverage Kubernetes service accounts
(<<{p}-gke-workload-identiy,GKE Workload Identity>> or <<{p}-iam-service-accounts,AWS IAM roles for service accounts>>, for example).